awscli/argparser.py

```diff
@@ -59,6 +60,11 @@ def choices(self, val):
         pass
 
 
+class DisablePagerAction(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        os.environ['AWS_PAGER'] = ''
```

Comment:
Ah so I'd recommend following this PR for implementing this flag: https://github.com/aws/aws-cli/pull/4748

Specifically we should define it in `awscli/data/cli.json` so the argument gets auto generated and we can then register a handler that set the session variable for the pager.